Alert now comes with optionnal explanatory messages

DebugBill opened the pull request:
SendAlertSensor now needs one more text parameter (only two devices are using this call, so it was easy to update)
If a text is provided, it appears on the GUI next to the numeric level. This allows for the 4 alarm levels to be reported with a explanation/description of the issue
If no text is provided, the behaviour is as before

I've tested with notifications, it did not break anything.
Did not have to modify the GUI code as it already relied on the string database field.

This makes the Text sensor even less usefull as one does not need to use both Alert and Text to report a detailed alarm.

DebugBill committed:
Merge branch 'master' of https://github.com/domoticz/domoticz

DebugBill committed:
Problem with Git and CRLF

DebugBill committed:
First commit, maybe does not compile

DebugBill committed:
Merge branch 'master' of https://github.com/domoticz/domoticz into AlertWithMessages

DebugBill committed:
Fix compile issue

DebugBill committed:
SendAlertSensor now accepts a message along with the alert level. The messages appears on the GUI
If no message is provided, the numeric level is displayed instead.
This has no impact on notifications.

DebugBill committed:
Better message for Multifun

DebugBill committed:
Moved some code from DomoticzHardware to MainWorker so it applies to all call to
alertSensors, not only the ones using standard helper.

gizmocuz commented:
Looks we have a conflict because of a recent pull request merge on ecodevices (i think it was yours...)

gizmocuz commented:
Do you have a screenshot of the new sensors usage ?
No need to update all current text sensors ?

DebugBill commented:
Here is a screen shot (hopes it goes through GitHub)

It needed an update of all calls to SendAlertSensor…. But there were only 3 throughout the code.
For direct calls to sDecodeRX there is no impact (thanks to my latest change a few minutes ago. It was not true one hour ago)

[cid:image001.png@01D29EFC.4F843FA0]

DebugBill committed:
Fix compile issue

szczukot commented on hardware/MultiFun.cpp:
```diff
@@ -414,7 +414,7 @@ void MultiFun::GetRegisters(bool firstTime)
 					if (((m_LastAlarms != 0) != (value != 0)) || firstTime)
 					{
-						SendAlertSensor(0, 255, value ? 4 : 1, "Alarm");
+						SendAlertSensor(0, 255, value ? 4 : 1, value ? "Alarm" : "OK", "Alarm");
```
wrong. Send always "alarm" as text

DebugBill replied:
reverted to original behaviour

szczukot commented on hardware/MultiFun.cpp:
```diff
@@ -436,7 +436,7 @@ void MultiFun::GetRegisters(bool firstTime)
 					if (((m_LastWarnings != 0) != (value != 0)) || firstTime)
 					{
-						SendAlertSensor(1, 255, value ? 3 : 1, "Warning");
+						SendAlertSensor(1, 255, value ? 3 : 1, value ? "Warning" : "OK", "Alarm");
```
Wrong
Send always "Warning" as text

DebugBill replied:
reverted to original behaviour

szczukot commented on hardware/SatelIntegra.cpp:
```diff
@@ -771,7 +771,7 @@ void SatelIntegra::ReportZonesViolation(const int Idx, const bool violation)
 	m_zonesLastState[Idx - 1] = violation;
 
-	SendAlertSensor(Idx, 255, violation ? 3 : 1, NULL);
+	SendAlertSensor(Idx, 255, violation ? 3 : 1, "", "Alert");
```
I don't want set name for this sensor.
Must be NULL instead "Alert"

DebugBill replied:
reverted to original behaviour

DebugBill committed:
Fix bug, hope this is OK this time

gizmocuz commented:
Maybe because you are reply by email that the image does not work
Also you copy part of your email into this, which i have to remove

Could you copy/paste it directly in github

Ps. Our automatic build systems have issues, see below (on git)

DebugBill committed:
Merge branch 'master' of https://github.com/domoticz/domoticz into AlertWithMessages
Reverted changes to SatelIntegra and Multifun to original behaviour

Conflicts:
	hardware/EcoDevices.cpp

DebugBill commented:
Fixed the merge issue and reverted Multifun and SatelIntegra to original behaviour.

Here is a screenshot

Now waiting for successful build

![alerts](https://cloud.githubusercontent.com/assets/25805255/24036152/a05277ae-0af8-11e7-92c9-494e3ceacc9c.png)

DebugBill committed:
Fix compile

szczukot commented:
Idea with additional text is good.
Maybe i will try use it in Satel and MultiFun

DebugBill committed:
fix again.

DebugBill commented:
Please don't merge yet, I'm not happy with the current behaviour but I will not have time to look into it before the week-end

gizmocuz commented:
Okey, send me a PM on the forum

DebugBill committed:
Merge branch 'master' of https://github.com/domoticz/domoticz into AlertWithMessages

DebugBill committed:
Reverted sensors from kWhMeter to P1SmartMeter for EJP, Heures Creuses and Tempo.
Fixed Demain bug
Reduced the number of updates to 1 per mn to avoid clogging the logs and the database
Still problems with alerts

DebugBill committed:
Still not perfect

DebugBill commented:
This version fixes most (I think) of the negative feedback on the forum.
Still some issues with passing text on alerts, but it should not be a major problem.

I did not find any explanation to the crashes reported by some users, I will continue to investigate now that I have many testers.

gizmocuz commented:
Please forget the 'negativity' of the feedback... i (most times) consider them as anomalies ;)
